columnphysics/icepack_shortwave_data.F90

```diff
@@ -639,51 +659,51 @@ subroutine icepack_shortwave_init_snicartest()
          ! reshape from nmbrad,nspint to nspint,nmbrad with order = 2,1
 
          ssp_snwextdr = reshape((/ &
-               46.27374983, 24.70286257, 6.54918455, 2.6035624,  1.196168,   &
-               46.56827715, 24.81790668, 6.56181227, 2.60604155, 1.19682614, &
-               46.76114033, 24.90468677, 6.56950323, 2.60780948, 1.19737512, &
-               46.9753992,  24.9495155,  6.57687695, 2.60937383, 1.19774008, &
-               47.48598349, 25.14100194, 6.59708024, 2.61372576, 1.19897351 /), &
+               46.27374983_dbl_kind, 24.70286257_dbl_kind, 6.54918455_dbl_kind, 2.60356240_dbl_kind, 1.19616800_dbl_kind,   &
+               46.56827715_dbl_kind, 24.81790668_dbl_kind, 6.56181227_dbl_kind, 2.60604155_dbl_kind, 1.19682614_dbl_kind, &
```

Comment:
Is declaring the variable as dbl_kind not sufficient?

Reply:
It isn't, which I found last week too.  Was debugging some data table stuff last week.  What happens is the explicit number in the data tables is single by default, it then get promoted to double when it's copied into the variable.  So you potentially lose some precision and also have less control across different compilers/machines.  Basically, if var is a double,

var = 1.000000000001   =>  var = 1.00000087240589 (pseudo-random extra digits)
var = 1.000000000001_dbl_kind => var = 1.000000000001

We know that's true when setting values in variables.  The same is true in data tables which shouldn't be a surprise.  Even if we know values to only single precision, it's better to type constants as doubles explicitly in general.